jwst/stpipe/crds_client.py

```diff
@@ -38,6 +42,12 @@
 from crds.core import config, exceptions, heavy_client, log
 from crds.core import crds_cache_locking
 
+def get_exceptions_module():
+    """Provide external indirect access to the crds.core.exceptions module to
+    alleviate issues with circular dependencies.
+    """
+    return exceptions
+
```

Comment:
Is there a reason that one needs a function to return `exceptions`?  Since it is imported in `crds_client`, can't you just to the imports elsewhere as
```
>>> from jwst.stpipe.crds_client import exceptions
>>> raise exceptions.CrdsLookupError('foo')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
crds.core.exceptions.CrdsLookupError: foo
```

Reply:
CRDS and JWST CAL have circular imports;  it's a code smell but I don't think there's an easy fix without some mutual refactoring:   CRDS needs datamodels.   CAL needs best references.

When conflicts have arisen I've resolved them with hand rolled lazy imports.

get_exceptions_module() decouples import timing so that 
      timing(crds_client import) !- timing (exceptions import).   
exceptions is not imported until absolutely needed.    

I want to maintain the pattern that crds imports only happen in crds_client  to help mitigate what I'll dub "circular import spaghetti".   If all the interfaces to crds_client are kept as functions,  all the CRDS imports in crds_client can be made lazy later only updating that one module.